### PR TITLE
HOT-1518 Add validation to phone number

### PR DESCRIPTION
**Pull request by OlegKorniichuk**
### Jira Story

- [DocTool rule - Implement R - 05246 - Zero invalid as first digit] (https://osi-cwds.atlassian.net/browse/HOT-1518)

## Description
The phone number should not start from zero

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [ ] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: spec/javascripts/utils/phoneNumberValidatorSpec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    })
+    it('empty phone number', () => {
+      expect(getPhoneNumberErrors(''))
+        .toEqual(undefined)
+    })
+    it('valid phone number', () => {
+      expect(getPhoneNumberErrors('123'))
+        .toEqual(undefined)
+    })
+  })
```
Comment (pokharelmanish) [inline spec/javascripts/utils/phoneNumberValidatorSpec.js]: Good test coverage, we could also add a test case for `null` and `undefined`

**Comment by pokharelmanish** [comment]
retest the please

**Comment by pokharelmanish** [comment]
retest this please

File: app/javascript/utils/phoneNumberValidator.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const getPhoneNumberErrors = (phoneNumber) => {
+  if (phoneNumber && phoneNumber.match(/^\(?0.*$/)) {
+    return ['The phone number should not start from 0']
+  } else {
+    return undefined
```
Comment (wmitchell-cambria) [inline app/javascript/utils/phoneNumberValidator.js]: This is an odd return type. `getPhoneNumberErrors` either returns an array of errors, or undefined. Instead, consider making this an empty array, which would indicate that the validation was successful with no errors. Undefined makes it seem like we don't know if there are errors or not.

Comment (OlegKorniichuk) [reply app/javascript/utils/phoneNumberValidator.js]: Return type is fixed to be consistent

Comment (OlegKorniichuk) [reply app/javascript/utils/phoneNumberValidator.js]: Story is created
https://osi-cwds.atlassian.net/browse/HOT-2602

**Comment by OlegKorniichuk** [comment]
retest this please
